vita3k/interface.cpp

```diff
@@ -597,27 +626,136 @@ bool handle_events(EmuEnvState &emuenv, GuiState &gui) {
+                else if (event.key.keysym.scancode == emuenv.cfg.keyboard_button_down)
+                    gui::browse_pages_manual(gui, emuenv, SCE_CTRL_DOWN);
+            } else if (gui.vita_area.home_screen) {
+                if (event.key.keysym.scancode == emuenv.cfg.keyboard_button_left)
+                    gui::browse_home_apps_list(gui, emuenv, SCE_CTRL_LEFT);
+                else if (event.key.keysym.scancode == emuenv.cfg.keyboard_button_right)
+                    gui::browse_home_apps_list(gui, emuenv, SCE_CTRL_RIGHT);
+                else if (event.key.keysym.scancode == emuenv.cfg.keyboard_button_up)
+                    gui::browse_home_apps_list(gui, emuenv, SCE_CTRL_UP);
+                else if (event.key.keysym.scancode == emuenv.cfg.keyboard_button_down)
+                    gui::browse_home_apps_list(gui, emuenv, SCE_CTRL_DOWN);
+                else if (event.key.keysym.scancode == emuenv.cfg.keyboard_button_l1)
+                    gui::browse_home_apps_list(gui, emuenv, SCE_CTRL_L1);
+                else if (event.key.keysym.scancode == emuenv.cfg.keyboard_button_r1)
+                    gui::browse_home_apps_list(gui, emuenv, SCE_CTRL_R1);
+                else if (event.key.keysym.scancode == emuenv.cfg.keyboard_button_circle)
+                    gui::browse_home_apps_list(gui, emuenv, SCE_CTRL_CIRCLE);
+                else if (event.key.keysym.scancode == emuenv.cfg.keyboard_button_cross)
+                    gui::browse_home_apps_list(gui, emuenv, SCE_CTRL_CROSS);
+            } else if (gui.vita_area.live_area_screen) {
+                if (event.key.keysym.scancode == emuenv.cfg.keyboard_button_left)
+                    gui::browse_live_area_apps_list(gui, emuenv, SCE_CTRL_LEFT);
+                else if (event.key.keysym.scancode == emuenv.cfg.keyboard_button_right)
+                    gui::browse_live_area_apps_list(gui, emuenv, SCE_CTRL_RIGHT);
+                else if (event.key.keysym.scancode == emuenv.cfg.keyboard_button_up)
+                    gui::browse_live_area_apps_list(gui, emuenv, SCE_CTRL_UP);
+                else if (event.key.keysym.scancode == emuenv.cfg.keyboard_button_down)
+                    gui::browse_live_area_apps_list(gui, emuenv, SCE_CTRL_DOWN);
+                else if (event.key.keysym.scancode == emuenv.cfg.keyboard_button_l1)
+                    gui::browse_live_area_apps_list(gui, emuenv, SCE_CTRL_L1);
+                else if (event.key.keysym.scancode == emuenv.cfg.keyboard_button_r1)
+                    gui::browse_live_area_apps_list(gui, emuenv, SCE_CTRL_R1);
+                else if (event.key.keysym.scancode == emuenv.cfg.keyboard_button_circle)
+                    gui::browse_live_area_apps_list(gui, emuenv, SCE_CTRL_CIRCLE);
+                else if (event.key.keysym.scancode == emuenv.cfg.keyboard_button_cross)
+                    gui::browse_live_area_apps_list(gui, emuenv, SCE_CTRL_CROSS);
+            }
```

Comment:
this can be refactored. function like this 
```
int get_sce_ctrl_code_from_scancode(){if (event.key.keysym.scancode == emuenv.cfg.keyboard_button_left)
return SCE_CTRL_LEFT); ...
```

can be made and then use it like 
```
int ctrl_code =  get_sce_ctrl_code_from_scancode(); if (ctrl_code!=0)   gui::browse_live_area_apps_list(gui, emuenv, ctrl_code);
```
code is just an illustration. it have bugs and wrong spelling and wrong naming and so on

Reply:
ah yes good idea, i have try thinks for this,n compare controller no have maping so, goo idea